### PR TITLE
Enable py2docfx docs gen tool, remove the dockerimage docs validation

### PR DESCRIPTION
**Pull request by JimSuplizio**
This is the initial PR to enable py2docfx, a new docs generation tool that'll be generating our Learn docs. This also removes our dependency on the docker image for docs validation in our pipelines.

The old [PR](https://github.com/Azure/azure-sdk-for-python/pull/34175) was closed due to the fact it aged 9 months while waiting for the py2docfx tool to be updated and the files I'm updating had a number of edits.

This was tested and verified [python - template](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4287599&view=logs&j=19b87903-f8bc-5cbb-22d9-4f6eb3469d8e&t=01c0d070-1636-5aed-84c3-1f03a1861eaf), to test the actual release code path from a manual run and [python - core](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4287684&view=results) with SetDevVersion to test the release code path for scheduled builds.

### CODE REVIEW

File: eng/scripts/Language-Settings.ps1 [hunk 1]
```diff
@@ -427,36 +427,6 @@ function Import-Dev-Cert-python
...
-      -DocValidationImageId $DocValidationImageId
-
-    if (!$result) {
-      $allSucceeded = $false
-    }
-  }
-
-  return $allSucceeded
-}
-
```
Comment (danieljurek) [inline eng/scripts/Language-Settings.ps1:459]: Thank you for removing this duplicate.

File: eng/scripts/docs/Docs-Onboarding.ps1 [hunk 2]
```diff
@@ -67,153 +67,144 @@ function Get-python-DocsPackagesAlreadyOnboarded($docRepoLocation, $moniker) {
...
+#         }
+#       ]
+#   }
+# The above example contains the common settings for Python track 2 libraries.
+# There are a few things of note:
+# 1. install_type can be something other than pypi. source_code or dist_file are two examples of this
+#    but those are for track 1 or libraries released by other teams and not through our engineering system.
+# 2. extra_index_url only needs to exist on the object if PackageSourceOverride is set
+# 3. The reason this needs to be done using a json file instead of just a command line is because py2docfx
+#    doesn't handle the autodoc_default_options on the
```
Comment (scbedd) [inline eng/scripts/docs/Docs-Onboarding.ps1:100]: nit: comment trailing off.

**Review by scbedd — APPROVED** [review]
LGTM. Left a comment on a nit.

File: eng/scripts/docs/Docs-Onboarding.ps1 (hunk 2 above)
Comment (JimSuplizio) [reply eng/scripts/docs/Docs-Onboarding.ps1]: Finished the sentence in the 12th and final commit.

**Comment by JimSuplizio** [comment]
The 12th and final commit was only to fix a sentence fragment in the comments. The [pull-request](https://dev.azure.com/azure-sdk/public/_build/results?buildId=4287596&view=results) pipeline on the 11th commit, which actually code changes, ran just fine. I'm going merge this without waiting for the pull-request pipeline to run since there's no actual code changes.

**Pull request merged**